src/OpenTelemetry.Api/CHANGELOG.md

```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+* **Breaking change:** `Baggage.Current` behavior changed to match `AsyncLocal`
+  (copy-on-write). Changes now create a new
```

Comment:
Instead of "match AsyncLocal", consider "align with the specification" and link to https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md#overview _"The Baggage container MUST be immutable, so that the containing Context also remains immutable."_.

Reply:
Updated. How's that?

Reply:
Looks great! Thanks!